docs/data/joy/getting-started/templates/TemplateCollection.js

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import LZString from 'lz-string';
+import startCase from 'lodash/startCase';
+import NextLink from 'next/link';
+import { useTheme } from '@mui/joy/styles';
+import AspectRatio from '@mui/joy/AspectRatio';
+import Box from '@mui/joy/Box';
+import Card from '@mui/joy/Card';
+import List from '@mui/joy/List';
+import ListDivider from '@mui/joy/ListDivider';
+import Button from '@mui/joy/Button';
+import Typography from '@mui/joy/Typography';
+import SvgIcon from '@mui/joy/SvgIcon';
+import Visibility from '@mui/icons-material/Visibility';
+import codeSandbox from 'docs/src/modules/sandbox/CodeSandbox';
+import extractTemplates from 'docs/src/modules/utils/extractTemplates';
+
+const cache = {};
+const req = require.context('!raw-loader!./', true, /^\.\/[^/]+\/.*\.(js|tsx)$/);
+req.keys().forEach((key) => {
+  cache[key] = req(key);
+});
+
+function compress(object) {
+  return LZString.compressToBase64(JSON.stringify(object))
+    .replace(/\+/g, '-') // Convert '+' to '-'
+    .replace(/\//g, '_') // Convert '/' to '_'
+    .replace(/=+$/, ''); // Remove ending '='
+}
+
+function addHiddenInput(form, name, value) {
+  const input = document.createElement('input');
+  input.type = 'hidden';
+  input.name = name;
+  input.value = value;
+  form.appendChild(input);
+}
```

Comment:
Would it be better to create a shared abstraction for this? It's none trivial, and yet we duplicate it.